docs/deployment/getting-started-locally/getting-started-locally.md

```diff
@@ -0,0 +1,196 @@
+
+This document will guide you on how to setup `etcd-druid` on your local machine and how to provision and manage `Etcd` cluster(s).
+
+## 00-Prerequisites
+
+Before we can setup `etcd-druid` and use it to provision `Etcd` clusters, we need to prepare the development environment. Follow the [Prepare Dev Environment Guide](../../development/prepare-dev-environment.md) for detailed instructions.
+
+## 01-Setting up KIND cluster
+
+`etcd-druid` uses [kind](https://kind.sigs.k8s.io/) as it's local Kubernetes engine. The local setup is configured for kind due to its convenience only. Any other Kubernetes setup would also work. 
+
+```bash
+> make kind-up
+```
+
+This command sets up a new Kind cluster and stores the kubeconfig at `./hack/kind/kubeconfig`.  Additionally, this command also deploys a local container registry as a docker container. This ensures faster image push/pull times. The local registry can be accessed as `localhost:5001` for pushing and pulling images. 
+
+To target this newly created cluster, set the `KUBECONFIG` environment variable to the kubeconfig file.
+
+```bash
+> export KUBECONFIG=$PWD/hack/kind/kubeconfig
+```
+
+> **Note:**  If you wish to configure kind cluster differently then you can directly invoke the script and check its help to know about all configuration options.
+>
+> ```bash
+> > ./hack/kind-up.sh -h
+>   usage: kind-up.sh [Options]
+>   Options:
+>     --cluster-name  <cluster-name>   Name of the kind cluster to create. Default value is 'etcd-druid-e2e'
+>     --skip-registry                  Skip creating a local docker registry. Default value is false.
+>     --feature-gates <feature-gates>  Comma separated list of feature gates to enable on the cluster.
+> ```
+
+## 02-Setting up etcd-druid
+
+### Configuring etcd-druid
+
+Prior to deploying `etcd-druid`, it can be configured via CLI-args and environment variables. 
+
+* To configure CLI args you can modify [`charts/druid/values.yaml`](https://github.com/gardener/etcd-druid/blob/3383e0219a6c21c6ef1d5610db964cc3524807c8/charts/druid/values.yaml).  For e.g. if you wish to `auto-reconcile` any change done to `Etcd` CR then you should set `enableEtcdSpecAutoReconcile` to true. By default this will be switched off.
+* `DRUID_E2E_TEST=true` : sets specific configuration for etcd-druid for optimal e2e test runs, like a lower sync period for the etcd controller.
+
+### Deploying etcd-druid
+
+Any variant of `make deploy-*` command uses [helm](https://helm.sh/) and [skaffold](https://skaffold.dev/) to build and deploy `etcd-druid` to the target Kubernetes cluster. In addition to deploying `etcd-druid` it will also install the [Etcd CRD](https://github.com/gardener/etcd-druid/blob/3383e0219a6c21c6ef1d5610db964cc3524807c8/config/crd/bases/crd-druid.gardener.cloud_etcds.yaml) and [EtcdCopyBackup CRD](https://github.com/gardener/etcd-druid/blob/3383e0219a6c21c6ef1d5610db964cc3524807c8/config/crd/bases/crd-druid.gardener.cloud_etcdcopybackupstasks.yaml).
```

Comment:
```suggestion
Any variant of `make deploy-*` command uses [helm](https://helm.sh/) and [skaffold](https://skaffold.dev/) to build and deploy `etcd-druid` to the target Kubernetes cluster. In addition to deploying `etcd-druid` it will also install the [Etcd CRD](https://github.com/gardener/etcd-druid/blob/3383e0219a6c21c6ef1d5610db964cc3524807c8/config/crd/bases/crd-druid.gardener.cloud_etcds.yaml) and [EtcdCopyBackupsTask CRD](https://github.com/gardener/etcd-druid/blob/3383e0219a6c21c6ef1d5610db964cc3524807c8/config/crd/bases/crd-druid.gardener.cloud_etcdcopybackupstasks.yaml).
```

Reply:
Done